src/k2/httpproxy/HTTPProxy.cpp

```diff
@@ -362,6 +362,125 @@ seastar::future<std::tuple<Status, CollectionCreateResponse>> HTTPProxy::_handle
     });
 }
 
+inline seastar::future<nlohmann::json> JsonResponse(Status&& status) {
+    nlohmann::json resp;
+    resp["status"] = status;
+    return seastar::make_ready_future<nlohmann::json>(std::move(resp));
+}
+
+inline seastar::future<nlohmann::json> JsonResponse(Status&& status, nlohmann::json&& response) {
+    nlohmann::json jsonResponse;
+    jsonResponse["status"] = status;
+    jsonResponse["response"] = std::move(response);
+    return seastar::make_ready_future<nlohmann::json>(std::move(jsonResponse));
+}
+
+// Get FieldToKeyString value for a type
+template <class T> void getEscapedString(SchemaField& field, nlohmann::json& jsonval,  String& out) {
+    T val;
+    (void)field;
+    if constexpr  (std::is_same_v<T, std::decimal::decimal64>
+        || std::is_same_v<T, std::decimal::decimal128>) {
+        throw k2::dto::TypeMismatchException("decimal type not supported with JSON interface");
+    } else {
+        jsonval.get_to(val);
+        out = FieldToKeyString<T>(val);
+    }
+}
+
+// Convert [{type: "FieldType", value: "value"}..] to string that can be used in
+// collection range end
+seastar::future<nlohmann::json> HTTPProxy::_handleGetKeyString(nlohmann::json&& request) {
+    String output;
+    if (!request.contains("fields"))
+        return JsonResponse(Status{400, "Invalid json"});
+
+    for (auto& record : request["fields"]) {
+        SchemaField field;
+        record.at("type").get_to(field.type);
+        String out;
+        K2_DTO_CAST_APPLY_FIELD_VALUE(getEscapedString, field, record["value"], out);
+        output += out;
+    }
+    return JsonResponse(Status{200, "success"}, nlohmann::json{{"result", output}});
```

Comment:
The key string can contain null and non printable bytes. We should check if the json library is handling that properly or if we need to encode it manually.

Reply:
That's good. What about the return path though? The intent is that the user can use this to find the range ends for the collection. Is the string decoded back to the format we expect for the collection creation api? You can make an issue for this if you want to save it for later.